moreorless/click.py

```diff
@@ -24,6 +24,24 @@ def echo_color_unified_diff(astr: str, bstr: str, filename: str, n: int = 3) ->
             click.secho(line, nl=False)
 
 
+def echo_color_precomputed_diff(diff: str) -> None:
+    """
+    Like `echo_color_unified_diff`, but for precomputed diff results.
+    """
+    for line in diff.splitlines(True):
+        # TODO benchmark and see if constructing the string up front is faster
+        if line.startswith("---") or line.startswith("+++"):
+            click.secho(line, bold=True, nl=False)
+        elif line.startswith("@@"):
+            click.secho(line, fg="cyan", nl=False)
+        elif line.startswith("-"):
+            click.secho(line, fg="red", nl=False)
+        elif line.startswith("+"):
+            click.secho(line, fg="green", nl=False)
+        else:
+            click.secho(line, nl=False)
```

Comment:
I don't like duplicating the code like this, but I also didn't want to a) make the signature for `echo_color_unified_diff()` more complicated, or b) end up making function calls for every single line of diff output. If you would prefer a or b to duplicating code, that's fine.

Reply:
Why don't you take line 13 and make that `echo_color_precomputed_diff(unified_diff(...))`